### ISSUE TITLE
Brave sync problem deleting devices and leaving the chain/creating new chain

### ISSUE THREAD

**Issue by nasatome** [brave/brave-browser#15362]
## Description 
Same issue: 
Brave sync problem deleting devices and leaving the chain/creating new chain
https://github.com/brave/brave-browser/issues/6349

## Steps to Reproduce

I noticed that my android device was not syncing with the chain and linux desktop, my android was syncing with my windows computer on the same laptop (dual boot) and I tried to sync my PopOS 20. 10 with a new computer just installed with Manjaro OS KDE (2021-04-18) and my PopOS computer appeared for a moment, but after a few minutes and I closed and opened brave in PopOS my chain changed and disappeared from the list of synced devices in my Manjaro. when I tried to press the "Leave Sync Chain" button it immediately generated a new chain. without asking me if I want to use a current one.

## Actual result:

https://user-images.githubusercontent.com/18271791/115159064-eedca000-a056-11eb-8a97-6a0e3f14975d.mp4

## Expected result:
that a new chain is not created each time I leave the current one

## Reproduces how often: 
No steps to reproduce

## Brave version (brave://version info)
Version 1.23.71 Chromium: 90.0.4430.72 (Official Build) (64-bit) in Linux PopOS 20.10 (Ubuntu Fork)

```

Brave | 1.23.71 Chromium: 90.0.4430.72 (Official Build) (64-bit)
-- | --
Revision | b6172ef8d07ef486489a4b11b66b2eaeed50d132-refs/branch-heads/4430@{#1233}
OS | Linux
JavaScript | V8 9.0.257.17
User Agent | Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/90.0.4430.72 Safari/537.36
Command Line | /opt/brave.com/brave/brave --enable-dom-distiller --disable-domain-reliability --no-pings --extension-content-verification=enforce_strict --extensions-install-verification=enforce --origin-trial-public-key=bYUKPJoPnCxeNvu72j4EmPuK7tr1PAC7SHh8ld9Mw3E=,fMS4mpO6buLQ/QMd+zJmxzty/VQ6B1EUZqoCU04zoRU= --sync-url=https://sync-v2.brave.com/v2 --lso-url=https://no-thanks.invalid --variations-server-url=https://variations.brave.com/seed --enable-features=AutoupgradeMixedContent,SafetyTip,WebUIDarkMode,PasswordImport,ReducedReferrerGranularity,PrefetchPrivacyChanges,LegacyTLSEnforced,DnsOverHttps --disable-features=IdleDetection,TabHoverCards,SharingQRCodeGenerator,DirectSockets,AutofillServerCommunication,TextFragmentAnchor,BraveDomainBlock,SubresourceWebBundles,SignedExchangePrefetchCacheForNavigations,AutofillEnableAccountWalletStorage,WebOTP,NotificationTriggers,LangClientHintHeader,NetworkTimeServiceQuerying,SignedExchangeSubresourcePrefetch,SafeBrowsingEnhancedProtection,SafeBrowsingEnhancedProtectionMessageInInterstitials --flag-switches-begin --flag-switches-end
Executable Path | /opt/brave.com/brave/brave
Profile Path | /home/nasatome/.config/BraveSoftware/Brave-Browser/Default
Variations | 7146a73c-3f4a17df


```

I honestly don't know what more information to provide, but at least if this ticket is closed, it remains as a history of the problem.

**Comment by nasatome** [comment]
just to add: in version Version 1.24.85 Chromium: 90.0.4430.212 (Official Build) (64-bit) the problem remains

**Comment by AlexeyBarabash** [comment]
This may happened when there is some error received from the server. Eventually the sync chain will be established.

**Comment by nasatome** [comment]
@AlexeyBarabash  I'm waiting for that to happen, but it looks like I'll have to figure out how to backup and sync my bookmarks some other way.

just to add: Version 1.25.68 Chromium: 91.0.4472.77 (Official Build) (64-bit) the problem remains

**Comment by jsecretan** [comment]
Hi @nasatome would you mind submitting your brave://sync-internals from the PopOS browser?  Through this form?

https://upload-support.brave.com/sync

That would give us some more info about what is happening

**Comment by nasatome** [comment]
solved in Version 1.33.106 Chromium: 96.0.4664.110 (Official Build) (64-bit)

every so often I would check to see if it had been fixed, and since I updated this last time it has been fixed.
